src/Iris/Interactive/Question.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+{- |
+Module                  : Iris.Interactive.Question
+Copyright               : (c) 2023 Dmitrii Kovanikov
+SPDX-License-Identifier : MPL-2.0
+Maintainer              : Dmitrii Kovanikov <kovanikov@gmail.com>
+Stability               : Experimental
+Portability             : Portable
+
+Asking Questions. Receiving answers.
+
+@since x.x.x.x
+-}
+module Iris.Interactive.Question (
+    yesno,
+    YesNo (..),
+    parseYesNo, -- export needed for testing
+) where
+
+import Control.Monad.IO.Class (MonadIO (..))
+import Control.Monad.Reader (MonadReader)
+
+import Data.Text (Text, toUpper)
+import qualified Data.Text.IO as TIO
+import System.IO (hFlush, stdout)
+
+import Iris.Cli.Interactive (InteractiveMode (..))
+import Iris.Env (CliEnv (..), asksCliEnv)
+
+parseYesNo :: Text -> Maybe YesNo
+parseYesNo t = case toUpper t of
+    "Y" -> Just Yes
+    "YES" -> Just Yes
+    "N" -> Just No
+    "NO" -> Just No
+    _ -> Nothing
+
+{- | Parsed as Yes: "Y", "YES" (lower- or uppercase)
+
+Parsed as No: "N", "NO" (lower- or uppercase)
+-}
+data YesNo
+    = No
+    | Yes
+    deriving stock
+        ( Show
+          -- ^ @since x.x.x.x
+        , Eq
+          -- ^ @since x.x.x.x
+        , Ord
+          -- ^ @since x.x.x.x
+        , Enum
+          -- ^ @since x.x.x.x
+        , Bounded
+          -- ^ @since x.x.x.x
+        )
+
+{- | Ask a yes/no question to stdout, read the reply from terminal, return an Answer.
+
+In case of running non-interactively, return the provided default
+
+Example usage:
+
+@
+app :: App ()
+app = do
+    answer <- Iris.yesno "Would you like to proceed?" Iris.Yes
+    case answer of
+        Iris.Yes -> proceed
+        Iris.No -> Iris.outLn "Aborting"
+
+@
+
```

Comment:
Yes also add an example of what will users see in the output when running this function